catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -150,20 +142,141 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
                     .build();
         }
 
-        Map<String, Object> registryMap = reportMapBuilder.buildRegistryMap(registryPackage);
+        try {
+            html = registryReportBuilder.getHtmlFromRegistryPackage(registryPackage,
+                    registryReportBuilder.REPORT);
+        } catch (IOException e) {
+            String message = "Error in compiling and applying report template.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
```

Comment:
Your error messages should match the Produces type.  I would recommend making an error html template that takes a message and return HTML instead of a string.  What you have now will work in browsers but it is not following the standard and could break another system if they try to parse the error message as HTML.